js/data.js

```diff
@@ -6,3 +6,41 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var $entry;
+var $photoimg = document.querySelector('#entry-img');
+var $photoURL = document.querySelector('#photo');
+var $form = document.querySelector('#input-form');
+
+$photoURL.addEventListener('input', handleInput);
+$form.addEventListener('submit', handleSubmit);
+
+function handleInput(event) {
+  $photoimg.setAttribute('src', event.target.value);
+}
+
+function handleSubmit(event) {
+  event.preventDefault();
+  var title = event.target.title.value;
+  var photo = event.target.photo.value;
+  var notes = event.target.notes.value;
+  $entry = { title, photo, notes };
+  $entry.EntryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.unshift($entry);
+  $photoimg.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $form.reset();
+}
```

Comment:
You should relocate all this code to `main.js`, the only code that should be in `data.js` is the data model and any localStorage-related code